### PR TITLE
Adding Challenge 28

### PR DESCRIPTION
**Pull request by puneeth072003**
### What kind of changes does this PR include?

-   [ ] Fixes or refactors
-   [x] A new challenge
-   [ ] Additional documentation
-   [ ] Something else

#### Description
Adding a new challenge 28 as asked in issue #423 completing the Part 1 of that issue 
-   [x]  a secret in a closed github issue ^^.

### Relations
issue #423

### Checklist:

-   [x] All the contributions made are solely the work of me and my co-authors
-   [x] I tested the changes in this PR (if applicable)
-   [x] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [x] The PR passes pre-commit hooks and automated tests

### CODE REVIEW

**Comment by puneeth072003** [comment]
@commjoen @bendehaan 
I still have to create a issue with the secret.

File: src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge28.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import java.util.List;
+
+import static org.owasp.wrongsecrets.RuntimeEnvironment.Environment.DOCKER;
+
+@Component
+@Order(28)
+public class Challenge28 extends Challenge {
+
+    private String SecretKey="asdfdpuYUIadbaoasdpa07kj2003";
```
Comment (commjoen) [inline src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge28.java:20]:
```suggestion
    private String SecretKey=new String(Base64.decode(new String(Base64.decode("WVhOa1ptUndkVmxWU1dGa1ltRnZZWE5rY0dFd04ydHFNakF3TXc9PQ=="))));;
```

File: src/main/resources/explanations/challenge28.adoc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+=== Secret In a closed github issue
+User reveals his `AWS Secret key` by mistake while describing conversation between him and his friend while discussing about his plans for upcoming days in a `GitHub issue` then realizes it and closes that issue.
```
Comment (commjoen) [inline src/main/resources/explanations/challenge28.adoc:2]:
```suggestion
A user reveals his `AWS Secret key` by mistake while describing conversation between him and his friend while discussing about his plans for upcoming days in a `GitHub issue` then realizes it and closes that issue.
```

File: src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge28Test.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.owasp.wrongsecrets.ScoreCard;
+import org.owasp.wrongsecrets.challenges.Spoiler;
+
+@ExtendWith(MockitoExtension.class)
+class Challenge28Test {
+
+    @Mock
+    private ScoreCard scoreCard;
+    private String SecretKey="asdfdpuYUIadbaoasdpa07kj2003";
```
Comment (commjoen) [inline src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge28Test.java:17]:
```suggestion
    private String secretKey=new String(Hex.decode("61736466647075595549616462616f617364706130376b6a32303033"));
```

File: src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge28.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    public Challenge28(ScoreCard scoreCard) {
+        super(scoreCard);
+        this.SecretKey = SecretKey;
```
Comment (commjoen) [inline src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge28.java:24]:
```suggestion
```

File: src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge28.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Override
+    public boolean canRunInCTFMode() {
+        return true;
+    }
+
+
+    @Override
+    public Spoiler spoiler() {
+        return new Spoiler(SecretKey);
```
Comment (commjoen) [inline src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge28.java:35]:
```suggestion
        return new Spoiler(secretKey);
```

File: src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge28.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    public boolean answerCorrect(String answer) {
+        return SecretKey.equals(answer);
```
Comment (commjoen) [inline src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge28.java:40]:
```suggestion
        return secretKey.equals(answer);
```

File: src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge28Test.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    @Test
+    void spoilerShouldRevealAnswer() {
+        var challenge = new Challenge28(scoreCard);
+
+        Assertions.assertThat(challenge.spoiler()).isEqualTo(new Spoiler(SecretKey));
```
Comment (commjoen) [inline src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge28Test.java:22]:
```suggestion
        Assertions.assertThat(challenge.spoiler()).isEqualTo(new Spoiler(secretKey));
```

File: src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge28Test.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Test
+    void rightAnswerShouldSolveChallenge() {
+        var challenge = new Challenge28(scoreCard);
+
+        Assertions.assertThat(challenge.solved(SecretKey)).isTrue();
```
Comment (commjoen) [inline src/test/java/org/owasp/wrongsecrets/challenges/docker/Challenge28Test.java:29]:
```suggestion
        Assertions.assertThat(challenge.solved(secretKey)).isTrue();
```

**Review by commjoen — CHANGES_REQUESTED** [review]
Hi @puneeth072003 ! Your first challenge! thank you!
Here are some feedback items based on the textual and linting results. You can, for each suggestion click on "commit suggestion" or "add suggestion to batch"in the files overview, and then quickly add the recommended changes.
After that, let's file the issue :D !
Love this!

**Comment by puneeth072003** [comment]
@commjoen Issue as been created at #694 can you please send me your part of the message.

**Review by commjoen — APPROVED** [review]
Lgtm! Thank you @puneeth072003 !
@bendehaan would you be able to do one more round of textual reviewing on it please?

**Review by bendehaan — APPROVED** [review]

**Pull request merged**